ocw/lib/eks.py

```diff
@@ -0,0 +1,90 @@
+import os
+import json
+import kubernetes
+import boto3
+from ocw.lib.provider import Provider
+from ocw.lib.k8s import clean_jobs
+
+TAG_IGNORE = 'pcw_ignore'
+
+
+class EKS(Provider):
+    __instances = {}
+    default_region: str = 'eu-central-1'
+
+    def __new__(cls, vault_namespace):
+        if vault_namespace not in EKS.__instances:
+            EKS.__instances[vault_namespace] = self = object.__new__(cls)
+            self.__eks_client = {}
+            self.__kubectl_client = {}
+            self.__cluster_regions = None
+            self.__aws_dir = None
+
+        return EKS.__instances[vault_namespace]
+
+    def __init__(self, namespace: str):
+        super().__init__(namespace)
+        self.create_credentials_file()
+
+    def aws_dir(self):
+        if self.__aws_dir is None:
+            self.__aws_dir = os.path.expanduser("~/.aws")
+        return self.__aws_dir
+
+    def list_regions(self):
+        if self.__cluster_regions is None:
+            regions_query = self.cmd_exec(
+                f"aws ec2 describe-regions --query 'Regions[].RegionName' --output json --region {EKS.default_region}")
+            self.__cluster_regions = json.loads(regions_query.stdout)
+        return self.__cluster_regions
+
+    def create_credentials_file(self) -> None:
+        creds_file = f"{self.aws_dir()}/credentials"
+
+        if not os.path.exists(creds_file):
```

Comment:
and do you need to check if file exists to decide about dir creation ? 

what will happen if we will keep it like this : 
```
if not os.path.exists(self.aws_dir()):
                os.mkdir(self.aws_dir())
```

Reply:
Yes, but my initial intention is to create the credentials if not exists, so was an indentation problem. In next commit will be fixed

Reply:
ok now it make sense :) . python :)